rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -595,25 +596,54 @@ private void saveNewUTXOs(BtcTransaction btcTx) throws IOException {
      */
     public void releaseBtc(Transaction rskTx) throws IOException {
 
-        //as we can't send btc from contracts we want to send them back to the sender
+        Coin value = rskTx.getValue().toBitcoin();
+        final RskAddress senderAddress = rskTx.getSender();
+        //as we can't send btc from contracts we want to send them back to the senderAddressStr
         if (BridgeUtils.isContractTx(rskTx)) {
             logger.trace("Contract {} tried to release funds. Release is just allowed from standard accounts.", rskTx);
-            throw new Program.OutOfGasException("Contract calling releaseBTC");
+            if (activations.isActive(ConsensusRule.RSKIP185)) {
+                emitRejectEvent(value, senderAddress.toHexString(), RejectedPegoutReason.CALLER_CONTRACT);
```

Comment:
Wasn't the next exception returning the funds to the address? We probably want to revert with a reason rather than emitting an event

Reply:
It could be more clear to reject the call. But the RSKIP mentions this case as emitting an event. 

With the above code (and RSKIP spec), I'm not sure if the calling contract could detect the REJECT, and it could be important to it. At least, it is usual in Ethereum world that the smart contract has a way to detect the rejection (ie, by reject the call, or returning a boolean false)

Reply:
The pre-existing out of gas exception was not returning funds as the precompiled contracts didn't revert transactions.
We've made a choice to generate the event to leave evidence of the "rejection" on chain, as we didn't have the choice to revert transactions at the moment of writing the RSKIP.
We will consider changing this to revert the transaction as a whole.
But for now I think this is out of scope. I'll validate it with @aeidelman